querybook/webapp/ui/Comment/Comments.tsx

```diff
@@ -0,0 +1,257 @@
+        ],
+    },
+    {
+        id: 2,
+        text: 'memewowmew',
+        uid: 10,
+        created_at: 1684662936,
+        updated_at: 1683662936,
+        reactions: [
+            { reaction: '🩵', uid: 10 },
+            { reaction: '🩵', uid: 11 },
+            { reaction: '🤍', uid: 10 },
+        ],
+        child_comments: [
+            {
+                id: 4,
+                text: 'this is my comment. this is my comment. this is my comment. this is my comment.',
+                uid: 11,
+                created_at: 1683662936,
+                updated_at: 1683662936,
+                reactions: [],
+            },
+            {
+                id: 31,
+                text: 'this is my comment. this is my comment. this is my comment. this is my comment.',
+                uid: 10,
+                created_at: 1683662936,
+                updated_at: 1683662936,
+                reactions: [
+                    { reaction: '🩵', uid: 10 },
+                    { reaction: '🩵', uid: 11 },
+                    { reaction: '🤍', uid: 10 },
+                ],
+            },
+        ],
+    },
+];
+
+interface IComment {
+    id: number;
+    // TODO: clean this
+    text: string | DraftJs.ContentState;
+    uid: number;
+    created_at: number;
+    updated_at: number;
+
+    child_comments?: IComment[];
+
+    reactions: IReaction[];
+}
+
+export interface IReaction {
+    reaction: string;
+    uid: number;
+}
+
+const initialCommentValue = DraftJs.ContentState.createFromText('');
+
+export const Comments: React.FunctionComponent = () => {
+    const userInfo = useSelector((state: IStoreState) => state.user.myUserInfo);
+
+    const [openThreadIds, setOpenThreadIds] = React.useState<Set<number>>(
+        new Set()
+    );
+    const [currentComment, setCurrentComment] =
+        React.useState<DraftJs.ContentState>(initialCommentValue);
+    const [editingCommentId, setEditingCommentId] =
+        React.useState<number>(null);
+
+    const handleEditComment = (
+        commentId: number,
+        commentText: DraftJs.ContentState
+    ) => {
+        setEditingCommentId(commentId);
+        setCurrentComment(commentText);
+    };
+
+    const handleCommentSave = React.useCallback(() => {
+        if (editingCommentId) {
+            // TODO: edit comment
+            setEditingCommentId(null);
+        } else {
+            // TODO: save comment
+        }
+        setCurrentComment(initialCommentValue);
+    }, [editingCommentId]);
+
+    const onEnterPress = React.useCallback(
+        (evt: KeyboardEvent) => {
+            if (
+                matchKeyMap(evt, {
+                    key: 'Enter',
+                    name: 'Comment',
+                })
+            ) {
+                handleCommentSave();
+            }
+        },
+        [handleCommentSave]
+    );
+
+    useEvent('keydown', onEnterPress);
+
+    const renderFlatCommentDOM = (comment: IComment) => (
+        <Comment
```

Comment:
instead of passing these props, why not just let Comment take a IComment object as prop?